universal-application-tool-0.0.1/app/services/question/ReadOnlyQuestionServiceImpl.java

```diff
@@ -8,19 +8,22 @@
 
 public final class ReadOnlyQuestionServiceImpl implements ReadOnlyQuestionService {
   private final ImmutableMap<String, ScalarType> scalars;
-  private final ImmutableMap<String, QuestionDefinition> questions;
+  private final ImmutableMap<Long, QuestionDefinition> questionsById;
+  private final ImmutableMap<String, QuestionDefinition> questionsByName;
```

Comment:
This should probably be questionsByPath since name is an attribute on question type that holds the display name.

Reply:
oo good catch!